### ISSUE TITLE
mcumgr serial interface does not work with CDC_ACM UART

### ISSUE THREAD

**Issue by pabigot** [zephyrproject-rtos/zephyr#27727]
If `CONFIG_MCUMGR=y` and `CONFIG_UART_MCUMGR_ON_DEV_NAME="CDC_ACM_0"` then `uart_mcumgr_isr()` will never exit once it is entered when a command is received over the serial connection.

The problem seems to be with `cdc_acm_irq_is_pending()` which in that context returns `true` because both `tx_ready` and `tx_irq_ena` are true, causing the loop in `uart_mcumgr_isr()` to spin forever.

**Comment by emob-nordic** [comment]
Could You provide information which sample You have build and how I can quickly reproduce this? I've tried to extend `./samples/subsys/mgmt/mcumgr/smp_svr/` sample with no success.

Was it build from `master` or any release? Have You checked other releases as well? I want to exclude recent `usb_enable()` changes.

**Comment by pabigot** [comment]
OK, I've reproduced, documented, and confirmed my results in: https://github.com/pabigot/zephyr/commits/repro/27727

Thanks for looking into this, it is a showstopper for an upgradable usb dongle.

**Comment by emob-nordic** [comment]
I think the issue there is USB backend executes `uart_mcumgr_isr()` from system work Q and enter never ending while loop which adds another item for system workQ. This `while` will never let the next item from system work Q to schedule the work and make it unable to finish the communication. I've played a bit with work Qs for both cdc and mcumgr but this seems not helping. I've managed to make it work by reworking the `uart_mcumgr_isr()` a bit but this seems unstable - it works good for short echoes but not for long ones. I'll try to find the solution but it seems that the fact that CDC_ACM using system work Q is not the only issue here.
